nav2_map_server/README.md

```diff
@@ -62,83 +136,336 @@ occupied_thresh: 0.65
+map_server_2d_n1:
+    ros__parameters:
+        yaml_filename: "./occupancy_grid_map.yaml"
+map_server_2d_n2:
+    ros__parameters:
+        yaml_filename: "./occupancy_grid_map.yaml"
+map_server_3d_n3:
+    ros__parameters:
+        yaml_filename: "./point_cloud_map.yaml"
+map_server_3d_n4:
+    ros__parameters:
+        yaml_filename: "./point_cloud_map.yaml"
+```
+
+- Map server
+```yaml
+# For composite MapSaver launch
+# examples/multi_map_saver_node/multi_composite_node_params.yaml
+map_saver_server_2d_1:
+    ros__parameters:
+        save_map_timeout: 2.0
+        free_thresh_default: 0.25
+        occupancy_thresh_default: 0.65
+map_saver_server_2d_2:
+    ros__parameters:
+        save_map_timeout: 2.0
+        free_thresh_default: 0.25
+        occupancy_thresh_default: 0.5
+map_saver_server_3d_3:
+    ros__parameters:
+        save_map_timeout: 5.0
+map_saver_server_3d_4:
+    ros__parameters:
+        save_map_timeout: 4.0
+```
+
+Then, one can invoke the process via launch file, with the params file that contains the parameters for both nodes:
+
+```shell
+# Files are there in the directory: `$Navigation2_Dir$/$nav2_map_server$/examples/multi_map_saver_node`
+$ ros2 launch multi_composite_map_saver.launch.py params_file:="multi_composite_node_params.yaml"
+```
+```shell
+# Files are there in the directory: `$Navigation2_Dir$/nav2_map_server/examples/multi_map_server_node`
+$ ros2 launch multi_composite_map_server.launch.py params_file:="multi_composite_node_params.yaml"
+```
 
-- loadMapYaml(): Load and parse the given YAML file
-- loadMapFromFile(): Load the image from map file and generate an OccupancyGrid
-- loadMapFromYaml(): Load the map YAML, image from map file and generate an OccupancyGrid
-- saveMapToFile(): Write OccupancyGrid map to file
+**Note:** *Use the `params_file` as the input to parameters to a node launch command.*
 
+---
+### Supported Map Types
+
+- Occupancy grid (nav_msgs/msg/OccupancyGrid)
+- PointCloud PointCloud2 (sensor_msgs/msg/PointCloud2)
+
+Presently, only two main map-types are supported one for `2D` and other for `3D`, still one can implement their own with full freedom to hold all the aspects of your new `map type`. 
+> one needs to implement or use an existing I/O system.
+
+> Create a specialization for your map type e.g:
+```c++
+template<>class MapServer<new_mapT> : public nav2_util::LifecycleNode{};
+```
+
+>  if a new type of service/message is required one can write and directly add the service with a callback in the *specialization*.
+
+For more information on adding a new map-type, simply analyse 
+```c++ 
+// 3D map_server
+include/nav2_map_server/map_server_core.hpp
+include/nav2_map_server/map_3d/map_server_3d.hpp
+src/map_3d/map_server_3d.cpp
+include/nav2_map_server/map_server.hpp
+
+// 3D map_saver
+include/nav2_map_server/map_saver_core.hpp
+include/nav2_map_server/map_3d/map_saver_3d.hpp
+src/map_3d/map_saver_3d.cpp
+include/nav2_map_server/map_saver.hpp
+``` 
```

Comment:
I think, this was already introduced in a "Extensible Framework" section. There is no need to repeat.
Additionally, we do not need to advise developers to make analysis. Instead of this, there would be a concrete step-by-step instructions how-to add a new map type. But this is going out-of-scope this documentation and current ticket.
So, please remove this.